Overhaul navigation pattern (#272)
pxblue/doc-it: src/docs/patterns/navigation.mdx

import DividerImage from '../../app/assets/design-patterns/navigation/Divider Image.png';

<!-- KEYWORDS: design pattern navigation drawer menu -->

# Navigation

<ImageGrid images={[Banner]} regularWidth />

Navigation refers to the way in which users access the different areas of your application and move from screen to screen. The most common forms of primary navigation are drawers and tabs.

<React.Fragment>
    <MaterialDesignDescription
        title={'Understanding Navigation'}
        url={'https://material.io/design/navigation/understanding-navigation.html#types-of-navigation'}
        description={`Read about Material Design's navigation fundamentals.`}
    />
    <MaterialDesignDescription
        title={'Navigation Drawer '}
        url={'https://material.io/components/navigation-drawer/'}
        style={{ marginRight: 0 }}
    />
    <Divider />
</React.Fragment>

# Drawer Navigation

A navigation drawer is a panel that is attached on the side of your application. These typically contain a [List](/patterns/lists) of primary pages in your application for the user to access quickly.

> <ImageGrid images={[NavigationBreakdownImg]} regularWidth />

Author: In the image, the divider does not go all the way to the right

The fonts are using gray 500. Should change that to black 500

Reviewer: fixed and updated

Most drawers have four main sections:

-   **Header**: contains the name of the company
-   **Subheader**: contains the product 
-   **Body**: contains a list of primary pages that users can navigate too

> -   **Footer**: contains company logo, version number and trademark information

Author: This definition of "header" and "subheader" is different from what we have in the code: https://pxblue-components.github.io/react-dev/?path=/info/components-drawer--get-read-me-story 

Would be nice to invent some other terminologies

Reviewer: adjusted and updated

There are several different styles of drawer navigation, each with slightly different behavior.

<DemoCard
    repository={'Drawer'}
    angular={'https://pxblue-components.github.io/angular/?path=/story/components-drawer--within-a-drawer-layout'}
    react={'https://pxblue-components.github.io/react/?path=/story/components-drawer--in-drawer-layout'}
/>

## Permanent

A permanent drawer is always visible and always open. The permanent drawer is typically full-width, as opposed to the rail variant (see below). Use this variation if you have:

-   five or more top-level pages
-   a nested navigation hierarchy
-   enough space to justify it (typically desktop-only applications)

> <ImageGrid images={[NavigationPermanentImg]} regularWidth />

Author: This image is using a right border rather than an elevation

## Rail

A navigation rail is a permanently collapsed drawer. The rail is able to show less information than a standard drawer, but has the advantage of a more compact, space-saving package. Use this variation if you have:

-   fewer than seven top-level pages
-   a flat navigation hierarchy
-   limited space

> <ImageGrid images={[NavigationRailImg]} regularWidth />

Author: The nav rail image is 56 px wide with label, inconsistent with our design specs. 

There are only two types of rails: 
- 72px wide, with an optional label text. The label has to be the `caption` typography. The label can hyphenate to a new line, but should never end with an ellipsis to clip off the text. When the label is present, the divider is required.
- 56px wide, with an optional tooltip on hover. No dividers on this one.

And there should usually be a blue 500 opacity 5% highlight background

## Persistent

A persistent drawer is an always-visible drawer that can be opened and closed (expanded or collapsed). In its opened state, it looks and behaves like a permanent drawer. It the closed state, it looks and behaves like a navigation rail. The drawer is typically opened and closed by clicking an icon in the header. It can also be opened momentarily on desktop applications by hovering with the mouse. Use this variation if you have:

-   fewer than seven top-level pages
-   primary pages with labels that require complex icons or may be too long to wrap effectivly
-   limited space but require a nested navigation hierarchy

<React.Fragment>
    <ImageGrid
        images={[CollapsibleImg]}
        caption={[
            'Figure 1 displays the normal collapsed state of the menu.',
            'Figure 2 displays the open state of the menu.',
        ]}
        regularWidth

>     />

Author: Prefer to see a gif for this

Author: These two groups images seem duplicate. 

Would be nice to see 
group 1: collapsed drawer, expanded drawer
group 2: the gif, showing how they transition to different states

    <ImageGrid

>         images={[PersistentImg, PermanentImg]}

Author: I saw a black stroke. Better crop that off

>         images={[PersistentImg, PermanentImg]}

Author: This image is using a right border rather than an elevation

        caption={
            'When the user hovers on the Navigation Rail, the side navigation will expand temporarily, revealing the text details of each navigation item; when the user moves their cursor away, the whole side navigation panel collapses.'
        }
    />
</React.Fragment>

## Temporary

A temporary drawer is invisible until it is opened — typically by pressing a button in the [App Bar](/patterns/appbar). This variation of the drawer appears on top of the main navigation, typically using a dark overlay to block interaction with the main content. Use this variation if you have:

-   a mobile application
-   extreme space limitations

<React.Fragment>
    <ImageGrid images={[NavigationTemporaryImg]} regularWidth />
    <Divider />
</React.Fragment>

# Tab Navigation

> Tab navigation uses a horizontal bar with buttons for each navigation location. These can appear at the top or bottom of your application. Tabs can be used in conjunction with drawer navigation as a way to navigate between sub-pages (e.g., you may have several sub-pages on your main settings page).

Author: > These can appear at the top or bottom of your application.

Show one image on the top, one image on the bottom

Author: > Tabs can be used in conjunction with drawer navigation as a way to navigate between sub-pages

Is bottom tab navigation also intended to be used this way? I think most bottom navigations are intended to replace the drawer

Reviewer: Updated

> <ImageGrid images={[NavigationTabsImg]} regularWidth />

Author: The focus of this image is currently on that drawer, not the tab

Author: Have at least one tab in active state

<React.Fragment>
    <MaterialDesignDescription title={'Tabs'} url={'https://material.io/components/tabs'} />
    <MaterialDesignDescription title={'Bottom Navigation'} url={'https://material.io/components/bottom-navigation'} />
</React.Fragment>

# Behaviors

## Nested Navigation

When using drawer navigation, you can have multiple levels of navigation hierarchy. These appear as expandable panels in the drawer. When using a hierarchical navigation in the drawer, you should try to limit yourself to now more than two levels.

> <ImageGrid images={[NestedItemImg, NestedItemStandardsImg]} regularWidth />

Author: ![image](https://user-images.githubusercontent.com/8997218/94948180-e81c8000-04ac-11eb-87a5-2779293f07a6.png)

The documentations here are inaccurate ("BG White 200" "Inactive" etc)

Reviewer: fixed

Author: ![image](https://user-images.githubusercontent.com/8997218/94962785-03938500-04c5-11eb-934b-58bb0b636ec1.png)
This _active item_ is body 1 but semibold, it rests on a 5% blue500 background

![image](https://user-images.githubusercontent.com/8997218/94962810-0e4e1a00-04c5-11eb-9cac-dc9eb425976f.png)
This is body 1 / subtitle 1, they are the same

![image](https://user-images.githubusercontent.com/8997218/94962831-1efe9000-04c5-11eb-9fb0-423ef2e257f2.png)
This is body 1 SB on top of a body 2

> ## Content-Cover vs. Content-Resize

Author: This section is kinda covered by the text above?

<ImageGrid
    images={[ContentCoverDesktopImg]}
    caption={[
        'Figure 1 displays the normal collapsed state of the menu for desktop.',
        'Figure 2 displays the open state of the menu with a gray overlay.',
    ]}
    regularWidth
/>
<ImageGrid
    images={[ContentCoverTabletImg]}
    caption={[
        'Figure 1 displays the normal collapsed state of the menu for tablet.',
        'Figure 2 displays the open state of the menu with a gray overlay.',
    ]}
    regularWidth
/>

## Dividers

Divider lines can be used between navigation items in the drawer. They are most effective when used for:

-   separating nested drawer items
-   separating sections of the drawer contents

<ImageGrid
    images={[DividerImage]}
    regularWidth
    caption={['Avoid using divider lines between every navigation item. These can make the drawer appear busy.']}
/>

# Design Specifications

> <ImageGrid images={[NavigationRailSpecsImg, NavigationRailLabeledImg, PanelSpecsImg, NavigationTabsSpecsImg]} regularWidth />

Author: The image above is also overusing the divider...

> <ImageGrid images={[NavigationRailSpecsImg, NavigationRailLabeledImg, PanelSpecsImg, NavigationTabsSpecsImg]} regularWidth />

Author: what about those "condensed" drawer items? (those with white margins in between, and highlight area 40px or 48px I can't remember)

> <ImageGrid images={[NavigationRailSpecsImg, NavigationRailLabeledImg, PanelSpecsImg, NavigationTabsSpecsImg]} regularWidth />

Author: ![image](https://user-images.githubusercontent.com/8997218/94948683-b2c46200-04ad-11eb-8013-ab7791cc36b0.png)
app bar should use H6
change one of the tab to active state
There is no "button 2" in the standard pxblue component library. You will have to explicitly list out what it is

import DividerImage from '../../app/assets/design-patterns/navigation/Divider Image.png';

<!-- KEYWORDS: design pattern navigation drawer menu -->

# Navigation

<ImageGrid images={[Banner]} regularWidth />

Navigation refers to the way in which users access the different areas of your application and move from screen to screen. The most common forms of primary navigation are drawers and tabs.

<React.Fragment>
    <MaterialDesignDescription
        title={'Understanding Navigation'}
        url={'https://material.io/design/navigation/understanding-navigation.html#types-of-navigation'}
        description={`Read about Material Design's navigation fundamentals.`}
    />
    <MaterialDesignDescription
        title={'Navigation Drawer '}
        url={'https://material.io/components/navigation-drawer/'}
        style={{ marginRight: 0 }}
    />
    <Divider />
</React.Fragment>

# Drawer Navigation

A navigation drawer is a panel that is attached on the side of your application. These typically contain a [List](/patterns/lists) of primary pages in your application for the user to access quickly.

> <ImageGrid images={[NavigationBreakdownImg]} regularWidth />

Author: DrawerHeader -> Header 
DrawerSubheader -> Subheader

Author: I also suggest using those 1️⃣  2️⃣  ... labels 
![image](https://user-images.githubusercontent.com/8997218/94961212-3b4cfd80-04c2-11eb-870e-276a3b4637ab.png)
for a better readability on mobile

Reviewer: not sure what this means or where the DrawerHeader/DrawerSubheader your referencing are

Most drawers have four main sections:

-   **Header**: contains the name of the company, name of the product or a logo
-   **Subheader**: optional section that can be used to support custom content, such as filtering or search

> -   **Body**: contains a list of primary pages that users can navigate too

Author: ```suggestion
-   **Body**: contains a list of primary pages that users can navigate to
```

-   **Footer**: contains company logo, version number and trademark information

There are several different styles of drawer navigation, each with slightly different behavior.

<DemoCard
    repository={'Drawer'}
    angular={'https://pxblue-components.github.io/angular/?path=/story/components-drawer--within-a-drawer-layout'}
    react={'https://pxblue-components.github.io/react/?path=/story/components-drawer--in-drawer-layout'}
/>

## Permanent

A permanent drawer is always visible and always open. The permanent drawer is typically full-width, as opposed to the rail variant (see below). Use this variation if you have:

-   five or more top-level pages
-   a nested navigation hierarchy
-   enough space to justify it (typically desktop-only applications)

> <ImageGrid images={[NavigationPermanentImg]} regularWidth />

Author: ![image](https://user-images.githubusercontent.com/8997218/94961549-d9d95e80-04c2-11eb-98f2-87d0ab824ac6.png)

This shadow is blurred... use the latest shadow in DSM

Reviewer: updarted

## Rail

A navigation rail is a permanently collapsed drawer. The rail is able to show less information than a standard drawer, but has the advantage of a more compact, space-saving package. Use this variation if you have:

-   fewer than seven top-level pages
-   a flat navigation hierarchy
-   limited space - recommended for HMI devices

> <ImageGrid images={[NavigationRailImg]} regularWidth />

Author: ![image](https://user-images.githubusercontent.com/8997218/94961683-1dcc6380-04c3-11eb-9f48-273f38282692.png)
avoid showing a menu icon here. The menu icon invokes a "drawer-opening" action, but a navigation rail aways stays collapsed.

Author: We can show two pics, one for the 72px rail, one for the 56px rail

## Persistent

> A persistent drawer is an always-visible drawer that can be opened and closed (expanded or collapsed). In its opened state, it looks and behaves like a permanent drawer. It the closed state, it looks and behaves like a navigation rail. The drawer is typically opened and closed by clicking an icon in the header. It can also be opened momentarily on desktop applications by hovering with the mouse. Use this variation if you have:

Author: > It the closed state

Typo

-   fewer than seven top-level pages

> -   primary pages with labels that require complex icons or may be too long to wrap effectivly

Author: > primary pages with labels that require complex icons or may be too long to wrap effectivly

I don't quite get what you mean by "complex icons" and "too long to wrap effectively"

Reviewer: complex icons meaning icons that are very difficult to understand the meaning at first look. for example we have a bunch of products that use very industry specific terminology and using stock icons doesnt work. in these cases we have to create new icons or use icons in places where they normally wouldnt make sense

Author: I see. What is "too long to wrap effectively" here?

-   limited space but require a nested navigation hierarchy

<React.Fragment>
    <ImageGrid
        images={[CollapsibleImg]}
        caption={[
            'Figure 1 displays the normal collapsed state of the menu.',
            'Figure 2 displays the open state of the menu.',
        ]}
        regularWidth
    />
    <ImageGrid
        images={[PersistentImg, PermanentImg]}
        caption={
            'When the user hovers on the Navigation Rail, the side navigation will expand temporarily, revealing the text details of each navigation item; when the user moves their cursor away, the whole side navigation panel collapses.'
        }
    />
</React.Fragment>

## Temporary

A temporary drawer is invisible until it is opened — typically by pressing a button in the [App Bar](/patterns/appbar). This variation of the drawer appears on top of the main navigation, typically using a dark overlay to block interaction with the main content. Use this variation if you have:

-   a mobile application
-   extreme space limitations

<React.Fragment>

>     <ImageGrid images={[NavigationTemporaryImg]} regularWidth />

Author: ![image](https://user-images.githubusercontent.com/8997218/94962356-515bbd80-04c4-11eb-9fa8-c1de5f22763c.png)

This is not a temporary drawer

Reviewer: then what is

Author: A temporary drawer always hovers above the content and usually accompanies a backdrop

    <Divider />
</React.Fragment>

# Tab Navigation

Tab navigation uses a horizontal bar with buttons for each navigation location. These can appear at the top or bottom of your application. On desktop and tablet, tabs at the top can be used in conjunction with drawer navigation as a way to navigate between sub-pages (e.g., you may have several sub-pages on your main settings page). Bottom tabs can be used in conjuction but is recommended that this pattern replaces the drawer navigation. On mobile it is also possible to combine top and bottom tabs on a single page.

> <ImageGrid images={[NavigationTabsImg]} regularWidth />

Author: Would be nice to have an image example of bottom navigation on mobile

Reviewer: updated

<React.Fragment>
    <MaterialDesignDescription title={'Tabs'} url={'https://material.io/components/tabs'} />
    <MaterialDesignDescription title={'Bottom Navigation'} url={'https://material.io/components/bottom-navigation'} />
</React.Fragment>

<Divider />

# Behaviors

## Nested Navigation

> When using drawer navigation, you can have multiple levels of navigation hierarchy. These appear as expandable panels in the drawer. When using a hierarchical navigation in the drawer, you should try to limit yourself to now more than two levels.

Author: ```suggestion
When using drawer navigation, you can have multiple levels of the navigation hierarchy. These appear as expandable panels in the drawer. When using hierarchical navigation in the drawer, you should try to limit yourself to no more than two levels.
```

<ImageGrid
    images={[ContentCoverDesktopImg]}
    caption={[
        'Figure 1 displays the normal collapsed state of the menu for desktop.',
        'Figure 2 displays the open state of the menu with a gray overlay.',
    ]}
    regularWidth
/>
<ImageGrid
    images={[ContentCoverTabletImg]}
    caption={[
        'Figure 1 displays the normal collapsed state of the menu for tablet.',
        'Figure 2 displays the open state of the menu with a gray overlay.',
    ]}
    regularWidth
/>

## Dividers

Divider lines can be used between navigation items in the drawer. They are most effective when used for:

-   separating nested drawer items
-   separating sections of the drawer contents

<ImageGrid
    images={[DividerImage]}
    regularWidth
    caption={['Avoid using divider lines between every navigation item. These can make the drawer appear busy.']}
/>

# Design Specifications

> <ImageGrid images={[NavigationRailSpecsImg, NavigationRailLabeledImg, PanelSpecsImg, NavigationTabsSpecsImg]} regularWidth />

Author: ![image](https://user-images.githubusercontent.com/8997218/94963036-7ac91900-04c5-11eb-923c-de84796f369b.png)

Drop that "open sans" (bc all our applications are using open sans) and do "14dp 600 / <text color>".  ("600" means "semibold")

Author: The app bar spec is already covered by https://pxblue.github.io/patterns/appbar#mobile, so we can remove the "H6" here

Reviewer: As a designer 14dp 600 means nothing to me. Are we expecting designers to understand that?

Reviewer 2: Can't we just use the name of the text style (e.g., subtitle2), or is this a one-off?

Author: yeah, the same holds for the developers... How about `14dp Semibold / 600`?

import DividerImage from '../../app/assets/design-patterns/navigation/divider.png';

<!-- KEYWORDS: design pattern navigation drawer menu tab tabs navigate nav -->

# Navigation

<ImageGrid images={[Banner]} regularWidth />

Navigation refers to how users access the different areas of your application and move from screen to screen. The most common forms of primary navigation are drawers and tabs.

<React.Fragment>
    <MaterialDesignDescription
        title={'Understanding Navigation'}
        url={'https://material.io/design/navigation/understanding-navigation.html#types-of-navigation'}
        description={`Read about Material Design's navigation fundamentals.`}
    />
    <Divider />
</React.Fragment>

# Drawer Navigation

A navigation drawer is a panel that is attached to the side of your application. These typically contain a [List](/patterns/lists) of primary pages in your application for the user to access quickly.

<ImageGrid
    images={[NavigationBreakdownImg]}
    caption={['1. Header', '2. Subheader', '3. Body', '4. Footer']}
    regularWidth
/>

Most drawers have four main sections:

> -   **Header**: contains the name of the company, name of the product, or a logo
> -   **Subheader**: an optional section that can be used to support custom content, such as filtering or search
> -   **Body**: contains a list of primary pages that users can navigate to
> -   **Footer**: contains the company logo, version number, and trademark information

Reviewer 2: ```suggestion
-   **Header**: typically contains the name of the company, name of the product, or a logo
-   **Subheader**: an optional section that can be used to support custom content, such as filtering or search
-   **Body**: contains a list of primary pages that users can navigate to
-   **Footer**: usually contains the company logo, version number, and trademark information
```

There are several different styles of drawer navigation, each with slightly different behavior.

<React.Fragment>
    <DemoCard
        repository={'Drawer'}
        angular={'https://pxblue-components.github.io/angular/?path=/story/components-drawer--within-a-drawer-layout'}
        react={'https://pxblue-components.github.io/react/?path=/story/components-drawer--in-drawer-layout'}
    />
    <MaterialDesignDescription
        title={'Navigation Drawer '}
        url={'https://material.io/components/navigation-drawer/'}
        style={{ marginRight: 0 }}
    />
</React.Fragment>

## Permanent

A permanent drawer is always visible and always open. The permanent drawer is typically full-width, as opposed to the rail variant (see below). Use this variation if you have:

-   five or more top-level pages
-   a nested navigation hierarchy
-   enough space to justify it (typically desktop-only applications)

> <ImageGrid images={[NavigationPermanentImg]} caption={['A permanent drawer on desktop.']} regularWidth />

Reviewer 2: Should the permanent drawer have an elevation, or should it be at the same level as the main content?

Author: I think we are trying to push for drawer w elevations for all variants

## Rail

> A navigation rail is a permanently collapsed drawer. The rail can show less information than a standard drawer but has the advantage of a more compact, space-saving package. Use this variation if you have:

Reviewer 2: ```suggestion
A navigation rail is a permanently collapsed drawer. The rail shows less information than a standard drawer but has the advantage of a more compact, space-saving appearance. Use this variation if you have:
```

> 

Reviewer 2: Should we say anything about the difference between condensed and normal rail?

Author: Mostly a matter of preference. Or we can push people to use the regular labeled version unless they are confident that their users can infer the icons’ meaning with no problem.

Reviewer 2: I'm okay with keeping it simple. I just noticed that we didn't say anything about it except in the caption of the image. Might be nice to add a sentence somewhere, e.g.: "The navigation rail can be regular (with text labels) or condensed (smaller area without labels)."

Author: I didn't see this before I hit that merge button 🙈 

Time to abuse your admin authority.

-   fewer than seven top-level pages
-   a flat navigation hierarchy
-   limited space - recommended for HMI devices

<ImageGrid images={[NavigationRailImg]} 
caption={[
        'Condensed Rail on desktop.',
    ]}
    regularWidth 
/>
<ImageGrid images={[NavigationRailLabelsImg]} 
caption={[
        'Standard rail with labels on desktop.',
    ]}
    regularWidth 
/>

## Persistent

A persistent drawer is an always-visible drawer that can be opened and closed (expanded or collapsed). In its open state, it looks and behaves like a permanent drawer. In its closed state, it looks and behaves like a navigation rail. The drawer is typically opened and closed by clicking an icon in the header. It can also be opened momentarily on desktop applications by hovering with the mouse. Use this variation if you have:

-   fewer than seven top-level pages

> -   primary pages with labels that require complex icons or too long to wrap effectively

Reviewer 2: ```suggestion
-   primary pages with complex icons or icons with long labels that wouldn't fit in a navigation rail
```

-   limited space but require a nested navigation hierarchy

<React.Fragment>
    <ImageGrid
        images={[CollapsibleImg]}
        caption={['1. Displays the normal collapsed state of the menu.', '2. Displays the open state of the menu.']}
        regularWidth
    />

>     <ImageGrid

Reviewer 2: Nice gif...do we want to speed it up to real-time (300ms i think?), or is it slow-mo to better show the interaction?

Author: I personally think slow motion shows the transition better.

<React.Fragment>
    <ImageGrid images={[NavigationTemporaryImg]} caption={'Opened temporary drawer with overlay.'} regularWidth />
    <Divider />
</React.Fragment>

# Tab Navigation

Tab navigation uses a horizontal bar with buttons for each navigation location. These can appear at the top or bottom of your application.

## Tabs on the Top

<ImageGrid
    images={[NavigationTabsImg]}
    caption={['A rail menu (#1) combined with a tab navigation (#2) to show sub-pages within the settings page.']}
    regularWidth
/>
<MaterialDesignDescription title={'Tabs'} url={'https://material.io/components/tabs'} />

You can combine tabs and a drawer as a way to navigate between sub-pages (e.g., you may have several sub-pages on your main settings page). In this case, the tabs usually sit on a secondary [app bar](/patterns/appbar).

## Tabs on the Bottom

<ImageGrid
    images={[NavigationBottomImg]}
    caption={[
        'Bottom tabs can be a good substitute for drawers when you have less than six primary navigation routes.',
    ]}
    regularWidth
/>
<MaterialDesignDescription title={'Bottom Navigation'} url={'https://material.io/components/bottom-navigation'} />

> For mobiles and tablets, you may use bottom tabs to replace the classical drawer navigations. Bottom tabs are a quick and easy way to navigate to the most frequently used primary navigation routes.

Reviewer 2: ```suggestion
For mobile and tablet applications, you may use bottom tabs to replace the classic drawer. Bottom tabs are a quick and easy way to navigate to the most frequently used primary navigation routes.
```

> Bottom tabs can also be used with a drawer. The tab is more prominent and can be used to highlight some main routes and features, while the drawer handles less important routes like settings, legal documents, FAQ pages.

Reviewer 2: ```suggestion
Bottom tabs can also be used in combination with a drawer. The tabs are more prominent and can be used to highlight some main routes and features, while the drawer handles the less important routes like settings, legal documents, FAQ pages.
```

<React.Fragment>
    <ImageGrid
        images={[NavigationBottomWithDrawerImg]}

>         caption={['A temporary content-cover menu (#1) combined with a bottom tab navigation (#2) on mobile.']}

Reviewer 2: ```suggestion
        caption={['1. Menu button activates a temporary content-cover menu', '2. Bottom tab navigation makes the primary pages more prominent.']}
```

        ]}
        regularWidth
    />
    <ImageGrid
        images={[ContentResizeImg]}
        caption={[
            '1. Displays the normal collapsed state of the menu.',
            '2. Displays the open state of the menu with content pushed to the right.',
        ]}
        regularWidth
    />
</React.Fragment>

## Dividers

Divider lines can be used between navigation items in the drawer. They are most effective when used for:

-   separating nested drawer items
-   separating sections of the drawer contents

<React.Fragment>
    <ImageGrid
        images={[DividerGroupImage]}
        regularWidth
        caption={['Dividers can be used to separate different sections of the drawer contents.']}
    />
    <ImageGrid
        images={[DividerImage]}
        regularWidth
        caption={[
            '1. Standard drawer navigation does not include dividers.',
            '2. Avoid using divider lines between every navigation item. These can make the drawer appear busy.',
        ]}
    />
    <Divider />
</React.Fragment>

# Design Specifications

## Mobile

<ImageGrid
    images={[
        PanelSpecsMobileImg,

>         NestedItemStandardsImg,

Reviewer 2: I don't remember, is the blue background of the drawer primary[50] as listed, or were we doing primary[500] with an opacity filter?

Reviewer 2: These specs also don't match our current implementation of the drawer. We may want to speed up our look into making those style changes.

Author: Yeah, in that default drawer discussion we decided the 5% opacity works better. Blue50 is a bit too heavy.

import TemporaryImg from '../../app/assets/design-patterns/navigation/navigation-temporary.png';
import PersistentImg from '../../app/assets/design-patterns/navigation/navigation-persistent.png';

<!-- New Images -->

import ProductExample1 from '../../app/assets/design-patterns/navigation/Product Example 1.png';
import ContentCover1 from '../../app/assets/design-patterns/navigation/Content Cover 1.png';
import ContentCover2 from '../../app/assets/design-patterns/navigation/Content Cover 2.png';
import GenericExample1 from '../../app/assets/design-patterns/navigation/Generic Example 1.png';
import NestedItemExample from '../../app/assets/design-patterns/navigation/Nested Item Example.png';
import NestedItemStandards from '../../app/assets/design-patterns/navigation/Nested Item Standards.png';
import ExpandableAccountInfo from '../../app/assets/design-patterns/navigation/Expandable Account Information.png';
import DividerImage from '../../app/assets/design-patterns/navigation/Divider Image.png';

<!-- KEYWORDS: design pattern navigation drawer menu -->

> # Navigation Patterns

Reviewer 2: ```suggestion
# Navigation
```

<ImageGrid images={[ProductExample1]} regularWidth />

Navigation is the way in which users access the different areas of your application. Main navigation is typically accomplished through a side drawer navigation panel with a list of links to the various pages in your application. PX Blue applications can have individual page or feature navigation as well. This can be accomplished by the use of tabs that can run across the top of your application.

> Keep in mind that these navigation patterns are tailored for PX Blue application use. PX Blue will be introducing traditional website navigation patterns in the near future.

Reviewer 2: Keep the intro short and sweet to introduce what navigation is. The detailed explanations of drawer/tab can go in sections below.

"Navigation is the way in which users are able to access the different parts of your application."

> 

Reviewer 2: Follow the section headers used in the other patterns: `Variations`

> ## Fixed Navigation

Reviewer 2: Should the top-level variations be "Drawer Navigtaion" and "Tab Navigation"? And then the four variations underneath drawer are Permanent, Persistent, Temporary, and Rail?

Reviewer: Split them out into separate sections

> The most basic navigation panel is fixed in place and always visible on the left side of the application The navigation can be a fixed list of icons with labels or a fixed panel that is open and revealing multiple navigation options Due to internationalization and accessibility standards PX Blue recommends the use of an open panel displaying the navigation options Another benefit of using a fixed navigation is the ability to use expandable tree navigation as well as more area for product branding. Note: A fixed navigation panel is not suitable when the application is viewed on smaller resolutions or mobile devices

Reviewer 2: I'm not exactly clear on what 'fixed icons with labels' and 'fixed panel' are...are we describing the Rail and the Permanent Drawer? We should match up the language.

>Do the NOTE as a callout

## Collapsible Navigation

> A more common implementation of the navigation panel is a collapsible panel that can be opened and closed via the menu icon in the top left corner of the application. This implementation can have two different variations: content-resize or content-cover.

Reviewer 2: Do we want to mention anything about hover behavior here, or are we anticipating taking that out of the component library?

Reviewer: If I can get an animation to work it is def worth putting in. It may be difficult to describe the desired behavior without some type of animation

Reviewer 2: I'm just talking about how (in the currently implementation) the persistent drawer will open temporarily when you hover over it. It was mentioned in a design review a while back that we might want to remove that behavior and just have click to open/close. Not sure where we landed on that.

INSERT IMAGE

### Content-Resize / Content-Push

<ImageGrid images={[PermanentImg]} />

The content-resize implementation does not cover any of the main application content. Rather, the navigation panel slides in and the main application container resizes to accommodate (the navigation panel and main panel are side by side instead of one on top of the other). This pattern should be considered for products with large amounts of data displays and application features.

### Content-Cover

<ImageGrid images={[ContentCover1, ContentCover2]} regularWidth />

> <ImageGrid images={[TemporaryImg]} />

Reviewer 2: Are these sections still relevant? Should we describe them under the variations that they apply to (temporary is content cover, the others are content-push)?

Reviewer: Moved to the variations section

The icon navigation bar (a.k.a. "[navigation rail](https://material.io/components/navigation-rail/#anatomy)" on Material.io) is a minimalist approach to the navigation panel. It features a single icon to represent each of the main areas of the application. This navigation panel is typically used on larger screens only, and is expandable to a full side navigation panel on hover or when clicking the menu icon.

These may be used in conjunction with Bottom Tab navigation - on larger resolution displays, bottom tabs would become elements in the side navigation.

> ## PX Blue Variations

Reviewer 2: This is more like a "Behavior" section

<ImageGrid images={[GenericExample1]} regularWidth />

> PX Blue provides the foundation for a wide range of navigation patterns. Ultimately the product will determine which pattern is best to use. The following sections provide PX Blue recommendations for common variation styles and standards on which variation to use in your product.

Reviewer 2: This doesn't really say anything. This whole page discusses PX Blue variations on navigation and where/when to use them.

Reviewer: Removed

### Nested Navigation Items

> PX Blue provides patterns to allow products to nest multiple layers of navigation options in the Navigation Menu. It is recommended not to exceed 2 levels of nested navigation. The graphic below illustrates the preferred style for nested navigation items.

Reviewer 2: "PX Blue provides patterns to allow products to" => "You can"

Reviewer 2: Generally, we spell out numbers less than ten, but I'm not sure how strictly we follow that.

Reviewer 2: "The graphic below illustrates" => Give the image a caption

<ImageGrid images={[NestedItemExample, NestedItemStandards]} regularWidth />

### Expandable Account Information

> PX Blue provides patterns for expandable account and settings within the navigation drawer. This feature can be found via a dropdown in expanded header area. It is recommended to use this pattern sparingly.

Reviewer 2: Are we actually recommending this pattern? The component library doesn't actually support it natively.

Reviewer: This can be removed

<ImageGrid images={[ExpandableAccountInfo]} regularWidth />

### Dividers

Dividers find the most usage in the Collapsible Navigation pattern. The reason for this is to help distinguish the clickable areas for the icons (collapsed state). In General avoid adding divider lines in random areas. However, there are some exceptions to the rules. For example, Footer, section header and nested nav items can use dividers to help break the content out from the rest of the list.

<ImageGrid images={[DividerImage]} regularWidth />

### Footer Behavior

> PX Blue recommends the use of a pinned footer in the navigation panel on all panel variations. An exception to this rule is if there is an extremely large menu structure. Exceptions can also be made if the product is using an expended header that is displaying a logo.

Reviewer 2: Need a specs and developers sections
